FairyTaleDefender/Assets/_Game/Scripts/Runtime/Entities/Weapons/Targeting/ScriptableObjects/RandomTargetTypeSO.cs

```diff
@@ -0,0 +1,23 @@
+using BoundfoxStudios.FairyTaleDefender.Extensions;
+using BoundfoxStudios.FairyTaleDefender.Infrastructure;
+using UnityEngine;
+
+namespace BoundfoxStudios.FairyTaleDefender.Entities.Weapons.Targeting.ScriptableObjects
+{
+	/// <summary>
+	/// Target type for selecting a random target.
+	/// </summary>
+	// We don't need more than one instance.
+	// [CreateAssetMenu(fileName = "RandomTargetType", menuName = Constants.MenuNames.Targeting + "/Random Target Type")]
+	public class RandomTargetTypeSO : TargetTypeSO
+	{
+		public override Collider GetTargetNonAlloc(Vector3 weaponPosition, NoAllocArrayResult<Collider> targets)
+		{
+			{
+				Debug.Assert(targets > 0, $"{nameof(targets.Size)} must be greater than 0.");
+
+				return targets.Result.PickRandom(targets)!;
```

Comment:
```suggestion
				return targets.Result.PickRandom(targets, targets.Size)!;
```

Oder? Sonst haben wir den #262 direkt wieder drin? :D

Reply:
Sollte eigentlich so passen :)
Zum einen haben wir aktuell keine Überladung mit 2 Parametern, zum anderen hast du einen impliziten Operator vom NoAllocArrayResult zu int, der die Size zurück gibt. Der Wert wird so als int aktuell auch dem Parameter mitgegeben :)
Können aber auch explizit targets.Size meinetwegen verwenden, um das ganze vllt etwas offensichtlicher zu machen.

Reply:
> Sollte eigentlich so passen :) Zum einen haben wir aktuell keine Überladung mit 2 Parametern, zum anderen hast du einen impliziten Operator vom NoAllocArrayResult zu int, der die Size zurück gibt. Der Wert wird so als int aktuell auch dem Parameter mitgegeben :) Können aber auch explizit targets.Size meinetwegen verwenden, um das ganze vllt etwas offensichtlicher zu machen.

Du hast vollkommen Recht. 
Ich hab das PickRandom auf das `targets` in der Klammer bezogen und nicht das `targets` vom `return`. :D